src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -519,9 +529,24 @@ def _compile_vts(self, vts, sources, analysis_file, upstream_analysis, classpath
 
         # If compiling a plugin, don't try to use it on itself.
         javac_plugins_to_exclude = (t.plugin for t in vts.targets if isinstance(t, JavacPlugin))
-        self.compile(self._args, classpath, sources, outdir, upstream_analysis, analysis_file,
-                     log_file, settings, fatal_warnings, zinc_file_manager,
-                     javac_plugins_to_exclude)
+        try:
+          self.compile(self._args, classpath, sources, outdir, upstream_analysis, analysis_file,
+                       log_file, settings, fatal_warnings, zinc_file_manager,
+                       javac_plugins_to_exclude)
+        except TaskError:
+          def find_failed_compile_log(workunits):
+            for workunit in workunits:
+              for output_name, outpath in workunit.output_paths().items():
+                # Locate failure log from child compilation workunit's stdout
+                if (workunit.name == self.name() and output_name == 'stdout'
```

Comment:
Matching `self.name` is a bit strange... a comment on why this is the appropriate thing to match would be good. Or maybe having an explicit property that matches the tool name?